### PR TITLE
Adds a short warm-up time when turning on a light

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

Adds a short semi-randomised delay after turning on a lightswitch before the actual lights themselves turn on.
Also adds sounds for light switches and lightbulbs flickering on.

This only happens when they're enabled manually from a light switch or APC, so it should act the same as it does now when using other methods.

*(The idea itself was taken from Skyrat, but the sounds and most of the code is original.)*

## Why It's Good For The Game

I don't think I've ever seen a light tube just turn on instantly in real life. More often than not they'll all do [this](https://youtu.be/73HkaOoBSjc) instead.
The changes in this PR aren't nearly as prominent as that example of course, but I still find that the effect it adds to the gameplay "immersion" is surprisingly substantial.
Primarily this just adds a nice little detail into the game. The sort of thing I can imagine someone being really impressed by when they first start playing.

When #15594 is merged this'll also be able to be made even fancier, with the light tube sprite flickering slightly before turning on fully.

## Images of changes

https://user-images.githubusercontent.com/57483089/129988373-355269a7-ea67-48c0-9885-dbbddf9d3121.mp4

## Changelog
:cl:
add: Added a slight delay after clicking a light switch for the room's lights to warm up before they turn on.
soundadd: Added a new sound for light switches.
soundadd: Added a new sound for light bulbs/tubes flickering on.
/:cl:

<hr>

- Light tube sound by mmaruska on [freesound.org](https://freesound.org/people/mmaruska/sounds/232447/), heavily modified.
- Light switch sound by AffectedArc07, also modified.

Both sounds can easily be tweaked further if necessary.

### CODE REVIEW

File: code/modules/power/lighting.dm [hunk 1]
```diff
@@ -290,62 +295,86 @@
...
-			static_power_used = brightness_range * 20 //20W per unit luminosity
+			static_power_used = brightness_range * 20 //20W per unit of luminosity
 			addStaticPower(static_power_used, STATIC_LIGHT)
 		else
 			removeStaticPower(static_power_used, STATIC_LIGHT)
 
+/**
+  * The actual proc to turn on the lightbulb.
+  *
+  * Do not call directly, use [/obj/machinery/light/proc/update] instead.
```
Comment (AffectedArc07) [inline code/modules/power/lighting.dm:332]: You may be able to use one of the spacemandmm helpers here to ensure you cant invoke this like a public method
```dm
	SHOULD_BE_PURE(TRUE)
	PRIVATE_PROC(TRUE)
	PROTECTED_PROC(TRUE)
```

One of these will work, see which one trips the vscode DM builder up
image

You can click this to test without doing the entire DM rebuild

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Requesting you also put the metadata back into the ogg I send

**Comment by AffectedArc07** [comment]
The ogg has the metadata, windows is just being an ass

File: code/modules/power/lighting.dm (hunk 1 above)
Comment (SabreML) [reply code/modules/power/lighting.dm:332]: Oh nice, I was wondering if it had something like that. Couldn't find any documentation for the spacemandmm stuff though so I just left it.

**Review by dearmochi — APPROVED** [review]

**Comment by TheYeetster1** [comment]
looks great I wonder how it will look with #16333

**Comment by Sinestia** [comment]
That's very pretty, nice work

**Review by variableundefined — APPROVED** [review]

**Review by Kyep — APPROVED** [review]

**Pull request merged**